test/electron/fixtures/app/main.js

```diff
@@ -43,6 +44,14 @@ function createWindow () {
   win.loadFile(join(__dirname, htmlRelativePath))
 }
 
+function makeSimpleGetRequest (fail = false) {
+  const url = fail ? 'https://non.existent.url/' : 'https://google.com/'
```

Comment:
I think we should use URLs we control if we're making requests for real, e.g. something like:

```suggestion
  const url = fail
    ? 'https://does.not.exist.bugsnag.com/'
    : 'https://bugsnag.com/'
```

Reply:
Maybe we should be hitting a local server though? That way the network conditions on CI shouldn't cause issues

Reply:
Yeah I went with those urls because that's what the Android e2e tests do, but this is a much better idea - will also be super useful when we come to add tests for response content length etc.

I've updated the tests to use the mock server and also created separate scenarios for a failed request (4xx response from the server) and a request error (hitting an invalid port on localhost)